src/main.rs

```diff
@@ -1,22 +1,52 @@
+use actix_web::{App, HttpServer};
+use futures_util::FutureExt;
 use std::env;
+use std::net::SocketAddr;
 use tokio::sync::Mutex;
+use tokio::try_join;
 use tonic::transport::Server;
 use tonic_health::server::health_reporter;
-use xenos::cache::RedisCache;
+use xenos::cache::redis::RedisCache;
+use xenos::metrics_server::metrics;
 use xenos::mojang::Mojang;
 use xenos::service::pb::profile_server::ProfileServer;
 use xenos::service::XenosService;
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     // read configuration from environment
-    let addr_str = env::var("SERVER_ADDR").unwrap_or("0.0.0.0:50051".to_string());
-    let redis_str = env::var("REDIS_ADDR").expect("redis address required");
+    let grpc_addr_str = env::var("SERVER_ADDR").unwrap_or("0.0.0.0:50051".to_string());
+    let redis_addr = env::var("REDIS_ADDR").expect("redis address required");
+    let metrics_addr = env::var("METRICS_ADDR").unwrap_or("0.0.0.0:3000".to_string());
```

Comment:
Let's use port 9990. It would theoretically collide with Wildfly, but I guess that can't be helped since all ports in the standard range are exhausted.